### PR TITLE
ECOM-5264 Make sure the banner image is stored at the correct path in S3

### PR DESCRIPTION
**Pull request by schenedx**
@rlucioni @clintonb Please review
This is to change the path where program banner image is stored.
Currently in stage, the program banner image is stored at `/program/<file_name>`
With this change, we will store the program image at `/media/programs/banner_images/<file_name>`

### CODE REVIEW

**Comment by clintonb** [comment]
:+1:

File: course_discovery/apps/course_metadata/utils.py [hunk 1]
```diff
@@ -24,3 +26,21 @@ def clean_query(query):
...
+    def __init__(self, populate_from, path, **kwargs):
+        self.populate_from = populate_from
+        self.path = path
+        super(UploadToFieldNamePath, self).__init__(populate_from, **kwargs)
+
+    def __call__(self, instance, filename):
+        field_value = getattr(instance, self.populate_from)
+        self.kwargs.update({
+            'name': field_value,
+            'path': self.path,
```
Comment (rlucioni) [inline course_discovery/apps/course_metadata/utils.py]: I'm a little confused about what's going on here. It looks like this is mapping some positional arguments into the keyword arguments expected by the constructor for `UploadTo`. Couldn't you just use `UploadTo`?

Comment (rlucioni) [inline course_discovery/apps/course_metadata/utils.py]: Oh, I think I see it. The field value is pulled off of the instance. Sorry, ignore me.

File: course_discovery/apps/course_metadata/tests/test_utils.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        ('/media', 'title', '.txt'),
+        ('', 'title', ''),
+    )
+    @ddt.unpack
+    def test_upload_to(self, path, field, ext):
+        upload_to = utils.UploadToFieldNamePath(populate_from=field, path=path)
+        upload_path = upload_to(self.program, 'name' + ext)
+        expected = os.path.join(path, str(getattr(self.program, field)) + ext)
+        print(upload_path)
+        print(expected)
```
Comment (rlucioni) [inline course_discovery/apps/course_metadata/tests/test_utils.py]: Remove these print statements.

**Comment by rlucioni** [comment]
👍

**Pull request merged**